Automattic/woocommerce-payments
# Cooldown – Optimize asset bundling and loading

#### Changes proposed in this Pull Request
Right now we load most of the assets by creating a JS React file for SVG images or importing them from the `/assets` folder, which result in base64 data URI. The issue with this approach is that some images get bundled multiple times, because we have multiple entries configured.

In addition, we also load some of those images from PHP and CSS files, so we can't just generate the new files on `dist/` as some of them will be duplicated.

This PR includes the following changes:
- Update webpack config to inline `svg/png` files by default. And generate a URL to files in `assets/` folders, for those imported with the query `?assets`. Using the same pattern, we already use for chunks and load assets from PHP, `[file]?ver=[hash]` as cache busting
- Add a new alias for `assets` to easily import them.
- Replace every JS SVG component with SVG files.
- Adjust SCSS rules replacing `svg` selectors with `img`.
- Remove unused assets, duplicated ones, and those with a Gridicons version.
- Optimize every SVG and PNG files.
- Update jest config to handles assets imported with query, and update every affected snapshot.

#### Testing instructions
As we don't have visual regression tests, and this PR affect many different components, the best way to test it is to visually check that there is no difference between `develop` and this branch. Checking all the images from WCPay (Overview, Settings, Transactions, etc) load correctly, without any markup difference.

>**Note**: Some payment method icons will be slighly different, because I unified them to avoid having multiple versions of the same.

-------------------


**Post merge**
